### ISSUE TITLE
no-fallthrough misses some statements

### ISSUE THREAD

**Issue by bartlomieju** [denoland/deno_lint#467]
```js
      case "symbol":
      case "function":
      default:
        throw new DOMException("Uncloneable value in stream", "DataCloneError");
```

```
error[no-fallthrough]: Fallthrough is not allowed
   --> /Users/biwanczuk/dev/deno/cli/rt/01_web_util.js:110:7
    |
110 |       case "function":
    |       ^^^^^^^^^^^^^^^^
    |
```

`case "symbol"` should be marked as well

**Comment by bartlomieju** [comment]
Actually I think `no-fallthrough` doesn't work as expected - it shouldn't produce diagnostics if there's no body for case statement.

**Comment by ross-weir** [comment]
> Actually I think `no-fallthrough` doesn't work as expected - it shouldn't produce diagnostics if there's no body for case statement.

Yeah it looks like this is the case. I'll have a shot at fixing this

**Comment by ross-weir** [comment]
If someone else wants to fix this go ahead, it's a bit outside my current level of knowledge. I'm not sure how to detect/handle [case statement bodies made up of only new lines](https://eslint.org/demo#eyJ0ZXh0IjoiZnVuY3Rpb24gdGVzdCgpIHtcblx0cmV0dXJuICdoaSdcbn1cblxuc3dpdGNoKCd0ZXN0Jykge1xuICBjYXNlICdhJzpcbiAgY2FzZSAnYic6XG5cblxuICBjYXNlICdjJzpcblx0dGVzdCgpO1xufSIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6NSwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6eyJjb25zdHJ1Y3Rvci1zdXBlciI6MiwiZm9yLWRpcmVjdGlvbiI6MiwiZ2V0dGVyLXJldHVybiI6Miwibm8tYXN5bmMtcHJvbWlzZS1leGVjdXRvciI6Miwibm8tY2FzZS1kZWNsYXJhdGlvbnMiOjIsIm5vLWNsYXNzLWFzc2lnbiI6Miwibm8tY29tcGFyZS1uZWctemVybyI6Miwibm8tY29uZC1hc3NpZ24iOjIsIm5vLWNvbnN0LWFzc2lnbiI6Miwibm8tY29uc3RhbnQtY29uZGl0aW9uIjoyLCJuby1jb250cm9sLXJlZ2V4IjoyLCJuby1kZWJ1Z2dlciI6Miwibm8tZGVsZXRlLXZhciI6Miwibm8tZHVwZS1hcmdzIjoyLCJuby1kdXBlLWNsYXNzLW1lbWJlcnMiOjIsIm5vLWR1cGUtZWxzZS1pZiI6Miwibm8tZHVwZS1rZXlzIjoyLCJuby1kdXBsaWNhdGUtY2FzZSI6Miwibm8tZW1wdHkiOjIsIm5vLWVtcHR5LWNoYXJhY3Rlci1jbGFzcyI6Miwibm8tZW1wdHktcGF0dGVybiI6Miwibm8tZXgtYXNzaWduIjoyLCJuby1leHRyYS1ib29sZWFuLWNhc3QiOjIsIm5vLWV4dHJhLXNlbWkiOjIsIm5vLWZhbGx0aHJvdWdoIjoyLCJuby1mdW5jLWFzc2lnbiI6Miwibm8tZ2xvYmFsLWFzc2lnbiI6Miwibm8taW1wb3J0LWFzc2lnbiI6Miwibm8taW5uZXItZGVjbGFyYXRpb25zIjoyLCJuby1pbnZhbGlkLXJlZ2V4cCI6Miwibm8taXJyZWd1bGFyLXdoaXRlc3BhY2UiOjIsIm5vLW1pc2xlYWRpbmctY2hhcmFjdGVyLWNsYXNzIjoyLCJuby1taXhlZC1zcGFjZXMtYW5kLXRhYnMiOjIsIm5vLW5ldy1zeW1ib2wiOjIsIm5vLW9iai1jYWxscyI6Miwibm8tb2N0YWwiOjIsIm5vLXByb3RvdHlwZS1idWlsdGlucyI6Miwibm8tcmVkZWNsYXJlIjoyLCJuby1yZWdleC1zcGFjZXMiOjIsIm5vLXNlbGYtYXNzaWduIjoyLCJuby1zZXR0ZXItcmV0dXJuIjoyLCJuby1zaGFkb3ctcmVzdHJpY3RlZC1uYW1lcyI6Miwibm8tc3BhcnNlLWFycmF5cyI6Miwibm8tdGhpcy1iZWZvcmUtc3VwZXIiOjIsIm5vLXVuZGVmIjoyLCJuby11bmV4cGVjdGVkLW11bHRpbGluZSI6Miwibm8tdW5yZWFjaGFibGUiOjIsIm5vLXVuc2FmZS1maW5hbGx5IjoyLCJuby11bnNhZmUtbmVnYXRpb24iOjIsIm5vLXVudXNlZC1sYWJlbHMiOjIsIm5vLXVudXNlZC12YXJzIjoyLCJuby11c2VsZXNzLWNhdGNoIjoyLCJuby11c2VsZXNzLWVzY2FwZSI6Miwibm8td2l0aCI6MiwicmVxdWlyZS15aWVsZCI6MiwidXNlLWlzbmFuIjoyLCJ2YWxpZC10eXBlb2YiOjJ9LCJlbnYiOnt9fX0=) while also handling no body for case statements.

I'll continue to experiment and see if I can figure it out.

**Comment by bartlomieju** [comment]
@ross-weir you can use [`SourceMap::span_to_snippet`](https://docs.rs/swc_common/0.10.4/swc_common/struct.SourceMap.html#method.span_to_snippet) to get a slice of source code. You can construct a span by taking `span.hi()` from `case` at question and `span.lo()` of following statement.
